### ISSUE TITLE
WorkItemDirectory & CorrelationPayloadDirectory

### ISSUE THREAD

**Issue by adamsitnik** [dotnet/performance#313]
Today I have tried to move all common build properties that are store in common.props file from `src\benchmarks\micro` to `src` but I have failed.

As far as I understand, we send to `Helix` two zipped folders:

- WorkItemDirectory: '$(Build.SourcesDirectory)\src\benchmarks\micro'
- CorrelationPayloadDirectory: '$(Build.SourcesDirectory)\scripts' 

https://github.com/dotnet/performance/blob/700816e0d2023e4263a9f3aa98a545fe0a2eeb76/eng/performance/benchmark_jobs.yml#L89-L91

In the Helix logs I can see that those two folders are copied to sth like:

```log
C:\dotnetbuild\work\82ac7b90-beba-445c-96a0-c3ed279fccb2\Work\e11567ec-0440-4e29-b6c0-e1f673b64baf\Exec - WorkItemDirectory
C:\dotnetbuild\work\82ac7b90-beba-445c-96a0-c3ed279fccb2\Payload\ - CorrelationPayloadDirectory
```

Because of this separation our logic in the Python script for finding the repo root fails because it simply assumes that root is scripFile.ParentDirectory

https://github.com/dotnet/performance/blob/d5c0a5cc438c9fa1a3e331fff41b60794f4cbf9c/scripts/performance/common.py#L87-L94

and we workaround it by providing two extra arguments to the Python script: `--bin-directory` and `--working-directory` 

The problem with the current solution is that we send only `src\benchmarks\micro` to Helix, so adding ML.NET/Scenarios would not work out of the box and require some extra work.

@adiaaida why do we need to send two separate zips to Helix? Why don't we just send the entire repo? 

I would like to get a better understanding of what we have today to make it easy to extend the repo with more benchmarks in the near futue

**Comment by adiaaida** [comment]
We cannot send the entire repo because send-to-helix/helixpublish write their logs to the artifacts dir at the same time that they would be trying to send the entire repo (including the artifacts dir) to helix. There is a lock on the file, so there is an exception.

What if you moved the scripts dir into the src dir?

**Comment by jorive** [comment]
Why don't we just pull the repo, build, run on the Helix machine instead? Building the harness is cheap.

**Comment by adiaaida** [comment]
I'll be honest, I'd really like us to not get into a place where we are doing things outside of how Helix is expected to operate.

However, that would be a pretty good solution for when we do this for the product repos and I'd like us to be as consistent as possible in those situations.

Also however: I don't think you can send a helix workitem with out a workitem directory. So, there are somethings you'd need to work around.

**Comment by jorive** [comment]
We could consider running the tests for this repo as build/run the tests. Thus, we would not need this extra step which does not buy anything.

**Comment by adiaaida** [comment]
What do you mean?

**Comment by jorive** [comment]
Sorry, what I meant to say is that splitting the build and run does not buy us anything here, but add extra complexity. BenchmarkDotNet needs the Cli to build the microbenchmarks anyways during the "run" step.

**Comment by adiaaida** [comment]
Right now, our infra does not split the build and run. It's all done on the Helix machines.

**Comment by billwert** [comment]
Fixed by #330
